src/UnoDrive.Shared/ViewModels/MyFilesViewModel.cs

```diff
@@ -83,40 +84,46 @@ public bool IsStatusBarLoading
 
 		public async void ItemClick(object sender, ItemClickEventArgs args)
 		{
-			if (args.ClickedItem is OneDriveItem driveItem)
+			if (args.ClickedItem is not OneDriveItem driveItem)
+				return;
+			
+			if (driveItem.Type == OneDriveItemType.Folder)
 			{
-				if (driveItem.Type == OneDriveItemType.Folder)
+				try
 				{
 					await LoadDataAsync(driveItem.Id, () =>
-					{
-						location.Forward = new Location
-						{
-							Id = driveItem.Id,
-							Back = location
-						};
-
-						location = location.Forward;
-					});
-
+								{
+									location.Forward = new Location
+									{
+										Id = driveItem.Id,
+										Back = location
+									};
+
+									location = location.Forward;
+								});
 				}
-				else
+				catch (Exception ex)
 				{
-					// TODO - open file
+					logger.LogInformation(ex, ex.Message);
 				}
 			}
+			else
+			{
+				// TODO - open file
+			}
 		}
 
 		void OnNetworkStatusChanged(object sender) =>
 			OnPropertyChanged(nameof(IsNetworkConnected));
 
-		async Task OnForwardAsync()
+		Task OnForwardAsync()
```

Comment:
if we just do one `await` operation we can return a `Task` and avoid creating a state machine (happens every time that you mark a method with the `async` keyword)

Reply:
TIL that it creates a state machine and it is better to just return the `Task`. Thanks for pointing this out!